site/content/blog/moving-existing-resources-to-a-new-app.md

```diff
@@ -0,0 +1,102 @@
+---
+title: "Moving existing resources to a new kapp app"
+slug: moving-existing-resources-to-a-new-kapp-app
+date: 2022-03-03
+author: Praveen Rewar
+excerpt: "Migrate from kubectl apply to kapp or move resources across apps"
+image: /img/logo.svg
+tags: ['carvel', 'kapp']
+
+---
+
+[kapp CLI](/kapp) encourages Kubernetes users to manage resources in bulk by working with "Kubernetes applications" (a set of resources with the same label). But how do we manage resources already present on the cluster (created by kubectl apply or are part of another kapp app)? 
+
+In this blog, we learn how to migrate from `kubectl apply` to a kapp app and move existing resources across kapp apps.
+
+## Migrating from `kubectl apply` to kapp
+Switching from kubectl apply to kapp deploy will allow kapp to adopt resources mentioned in a given config. However, kapp will try to insert a few of its labels in bodies of some resources, like Deployments, which may fail due to those resources having immutable fields that kapp tries to update (spec.selector on Deployments).
+
+```bash
+kapp: Error: Applying update deployment/server (apps/v1) namespace: default:
+  Updating resource deployment/server (apps/v1) namespace: default:
+    API server says:
+      Deployment.apps "server" is invalid: spec.selector:
+        Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app":"server", "kapp.k14s.io/app":"1645686658981326000"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable (reason: Invalid)
+```
+
+### Option 1
+To prevent this failure, add the [`kapp.k14s.io/disable-default-label-scoping-rules: ""`](/kapp/docs/latest/config/#labelscopingrules) annotation to individual resources (Deployments) to prevent kapp from touching the immutable fields when adopting them. Adding the annotation will exclude the resources from label scoping (used to scope resources within the current application).
+The annotation doesn't affect the app's ability to track such resources, the only consequence of disabling label scoping is that other resources in other apps cannot have the same label selector.
```

Comment:
Should we demonstrate the effect of using this annotation with an example?

Reply:
Sure, I will do it.

Reply:
Done.